fast/stages/1-vpcsc/main.tf

```diff
@@ -27,55 +59,29 @@ locals {
       }
     }
   }
-  perimeters = {
-    for k, v in var.perimeters : k => merge(v, {
-      restricted_services = (
-        v.restricted_services == null
-        ? local.restricted_services
-        : v.restricted_services
-      )
-      resources = distinct(concat(
-        v.resources,
-        k != "default" || var.resource_discovery.enabled != true ? [] : [
-          for v in module.vpc-sc-discovery[0].project_numbers :
-          "projects/${v}"
-        ]
-      ))
-    })
-  }
-  restricted_services = yamldecode(file("data/restricted-services.yaml"))
 }
 
 module "vpc-sc-discovery" {
   source           = "../../../modules/projects-data-source"
-  count            = var.resource_discovery.enabled == true ? 1 : 0
+  for_each         = var.resource_discovery.enabled ? local._perimeter_filters : tomap({})
   parent           = coalesce(var.root_node, "organizations/${var.organization.id}")
   ignore_folders   = var.resource_discovery.ignore_folders
   ignore_projects  = var.resource_discovery.ignore_projects
   include_projects = var.resource_discovery.include_projects
+  query            = each.value
 }
 
 module "vpc-sc" {
   source = "../../../modules/vpc-sc"
   # only enable if the default perimeter is defined
-  count         = var.perimeters.default == null ? 0 : 1
+  count         = length(local._data.perimeters) > 0 ? 1 : 0
   access_policy = var.access_policy
   access_policy_create = var.access_policy != null ? null : {
     parent = "organizations/${var.organization.id}"
     title  = "default"
   }
-  access_levels    = var.access_levels
```

Comment:
access levels should still be separate from perimeters, and referenced in perimeters by key

Reply:
same for ingress/egress policies